.github/workflows/pep8.yml

```diff
@@ -20,11 +20,14 @@ jobs:
       - uses: actions/setup-python@v4
         with:
           python-version: "3.10"
-        name: 'Setup python'
+        name: "Setup python"
       - shell: bash {0}
         run: |
-            python -m pip install --upgrade pip flake8
-        name: 'Install Flake8'
+          python -m pip install --upgrade pip flake8
+        name: "Install Flake8"
       - shell: bash {0}
         run: ./build_tools/flake8_diff.sh
-        name: 'Run Flake8 diff'
+        name: "Run Flake8 diff"
+      - shell: bash {0}
+        run: flake8 --config=setup.cfg --verbose "nilearn/decomposition"
+        name: "Run Flake8 on whole file"
```

Comment:
Added it as part of the already existing workflow.

Could have it as a separate one it this is more convenient.